daemon/snap.go

```diff
@@ -382,24 +382,25 @@ func mapRemote(remoteSnap *snap.Info) *client.Snap {
 		DownloadSize: remoteSnap.Size,
 		Icon:         snapIcon(remoteSnap),
 		ID:           remoteSnap.SnapID,
-		Name:         remoteSnap.Name(),
-		Revision:     remoteSnap.Revision,
-		Status:       status,
-		Summary:      remoteSnap.Summary(),
-		Type:         string(remoteSnap.Type),
-		Base:         remoteSnap.Base,
-		Version:      remoteSnap.Version,
-		Channel:      remoteSnap.Channel,
-		Private:      remoteSnap.Private,
-		Confinement:  string(confinement),
-		Contact:      remoteSnap.Contact,
-		Title:        remoteSnap.Title(),
-		License:      remoteSnap.License,
-		Screenshots:  screenshots,
-		Prices:       remoteSnap.Prices,
-		Channels:     remoteSnap.Channels,
-		Tracks:       remoteSnap.Tracks,
-		CommonIDs:    remoteSnap.CommonIDs,
+		// TODO parallel-install: revisit to properly include the name from the original query
```

Comment:
Not clear what this means here... the instance name holds the name of the snap too, but it probably means something else. Would be good to expand the comment to describe what the problem and candidate solution are.

Reply:
The TODO still seems to hint at a strange direction. This is not about including the name from the query. The query matches a snap, which can either be local, or remote, or both, or none. A local name may be an instance name, and a remote name can't. The query might include an instance key, or not. So the scenario is actually pretty simple: if the query has an instance key, it can only possibly match a local snap, and the resulting name should be the instance name, if any, or nothing. If the query misses an instance key, it's the current situation.

With that background, the TODO saying "include the name from the original query" seems awkward. We don't want to "include" anything else that is not already here, I believe.

So either the TODO needs to be improved so it more clearly indicates what the issue is, or it needs to be removed. This is a bit pedantic of my part given this is just an early PR, but I'd just like to make sure our discussion and consideration around this issue isn't lost.

Reply:
I think it's ok to drop the TODO. As the function name hints it, it's mapping a snap.Info that was assembled based on data from the store. Now, since we assume that `store` APIs will be aware of instance keys, whatever is returned there should already include instance key if that was passed in the first place, and `remoteSnap.InstanceName()` should give us the right 'name'.